### PR TITLE
Add Music lab progressSlider component

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR refactors the progress slider of the music lab play view that was added by https://github.com/code-dot-org/code-dot-org/pull/58293 and adds color to the progress bar.

Because `<input>` and `<progress>` elements need to be styled differently for different browsers and the styling was getting complicated, I opted to create a new small component `ProgressSlider` which displays the progress of the song being played with a slider and playhead. A different color is rendered on the bar to represent the percent progress.

https://github.com/code-dot-org/code-dot-org/assets/107423305/4a6dc2f8-8232-4b3d-ab12-ad8bf87de9a2

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/music/views/ProgressSlider.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+*/
+const ProgressSlider: React.FunctionComponent<ProgressSliderProps> = ({
+  percentProgress,
+}) => {
+  if (percentProgress < 0) {
+    percentProgress = 0;
+  } else if (percentProgress > 100) {
+    percentProgress = 100;
+  }
+  const sliderWidth = parseInt(moduleStyles.sliderWidth.replace('px', ''));
```
Comment (fisher-alice) [inline apps/src/music/views/ProgressSlider.tsx:18]: Importing `sliderWidth` variable from `progress-slider.module.scss`.

File: apps/src/music/views/ProgressSlider.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  
+*/
+const ProgressSlider: React.FunctionComponent<ProgressSliderProps> = ({
+  percentProgress,
+}) => {
+  if (percentProgress < 0) {
+    percentProgress = 0;
+  } else if (percentProgress > 100) {
+    percentProgress = 100;
+  }
```
Comment (thomasoniii) [inline apps/src/music/views/ProgressSlider.tsx]: This is fine as is, but for another suggestion:

```
const boundedPercentProgress = Math.min(Math.max(0, percentProgress), 100)
```

I kinda like that it _sorta_ reads in a row that `percentProgress` should be between `0` and `100`.

I'm not sure if this version adds enough clarity or removes enough redundancy to be worth it, though.

**Review by thomasoniii — APPROVED** [review]
trivial non-blocking range suggestion, but otherwise lgtm 🚀

File: apps/src/music/views/ProgressSlider.tsx (hunk 2 above)
Comment (fisher-alice) [reply apps/src/music/views/ProgressSlider.tsx]: I like it and I also like your variable name. Will update - thanks!

File: apps/src/music/views/ProgressSlider.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+interface ProgressSliderProps {
+  percentProgress: number;
+}
+
+/*
+  * A progress slider that displays the current progress of a music lab song.
+  * The slider is a gray bar that fills up with a white color as the song progresses.
+  * A playhead moves along the slider to show the current progress of the song.
+  * @param percentProgress - The percentage of the song that has been played.
+  * @param sliderWidth - The width of the slider in pixels.
```
Comment (sanchitmalhotra126) [inline apps/src/music/views/ProgressSlider.tsx]: nit: is `sliderWidth` meant to be an optional prop as well? (or if not, can we remove it from this doc?)

**Review by sanchitmalhotra126 — APPROVED** [review]
Nice refactor! The new component looks slick, and I can see it being useful outside Music Lab as well!

File: apps/src/music/views/ProgressSlider.tsx (hunk 3 above)
Comment (fisher-alice) [reply apps/src/music/views/ProgressSlider.tsx]: Thanks for catching! I originally had it as a prop, but removed and forgot to update comment. Will update!

**Pull request merged**